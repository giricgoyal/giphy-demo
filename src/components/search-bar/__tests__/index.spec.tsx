import React from 'react'
import { fireEvent, render } from '@testing-library/react'
import '@testing-library/jest-dom'

import SearchBar, { Props } from '../index'

describe('<SearchBar />', () => {
    let container, getByTestId
    const onChangeHandler = jest.fn()

    const defaultProps: Props = {
        value: '',
        onChange: onChangeHandler,
    }

    describe('When value is empty', () => {
        beforeEach(() => {
            ;({ container } = render(<SearchBar {...defaultProps} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })
    })

    describe('When value is provided', () => {
        let props: Props
        beforeEach(() => {
            props = {
                ...defaultProps,
                value: 'Happy',
            }
            ;({ container, getByTestId } = render(<SearchBar {...props} />))
        })

        test('should match snapshot', () => {
            expect(container).toMatchSnapshot()
        })

        describe('onChange', () => {
            describe('When input value is changed', () => {
                beforeEach(() => {
                    fireEvent.change(getByTestId('input'), {
                        target: {
                            value: 'Meh',
                        },
                    })
                })

                test('Should call onChangeHandler with correct value', () => {
                    expect(onChangeHandler).toHaveBeenCalledWith('Meh')
                })
            })

            describe('When input clear icon is clicked', () => {
                beforeEach(() => {
                    fireEvent.click(getByTestId('clear'))
                })

                test('should call onChangeHandler with ""', () => {
                    expect(onChangeHandler).toHaveBeenCalledWith('')
                })
            })
        })
    })
})
