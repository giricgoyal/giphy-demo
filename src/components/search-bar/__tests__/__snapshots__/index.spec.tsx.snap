// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> When value is empty should match snapshot 1`] = `
<div>
  <form
    class="search"
  >
    <input
      class="search__input form-control"
      data-testid="input"
      placeholder="Search"
      type="text"
      value=""
    />
    <svg
      class="search__icon icon"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
    
  </form>
</div>
`;

exports[`<SearchBar /> When value is provided should match snapshot 1`] = `
<div>
  <form
    class="search"
  >
    <input
      class="search__input form-control"
      data-testid="input"
      placeholder="Search"
      type="text"
      value="Happy"
    />
    <svg
      class="search__icon icon"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
    <svg
      class="search__clear icon"
      data-testid="clear"
    >
      <use
        xlink:href="#undefined"
      />
    </svg>
  </form>
</div>
`;
